### PR TITLE
fix: adds sort and filter functionality to add select columns

### PR DESCRIPTION
**Pull request by davidmenendez**
Contributes to #1752 

adds the filter and sort functionality to multi select with hierarchy columns.

Screen Shot 2022-03-09 at 3 37 07 PM
Screen Shot 2022-03-09 at 3 37 13 PM
Screen Shot 2022-03-09 at 3 37 20 PM

accomplishing this required a refactor of the items schema because previously there was no way (as far as i could tell) to support dynamically loading in sort and filter options. now the schema looks more like this

```
{
   "items":{
      "sortBy":[
         "title"
      ],
      "entries":[
         {
            "id":"1",
            "value":"folder 1",
            "title":"folder 1",
            "children":{
               "sortBy":[
                  "title",
                  "size"
               ],
               "filterBy":"fileType",
               "entries":[
                  {
                     "id":"1-1",
                     "value":"file1.pdf",
                     "title":"file1.pdf",
                     "fileType":"pdf",
                     "size":"100"
                  },
                  {
                     "id":"1-2",
                     "value":"index.js",
                     "title":"index.js",
                     "fileType":"js",
                     "size":"200"
                  },
                  {
                     "id":"1-3",
                     "value":"sitemap.xml",
                     "title":"sitemap.xml",
                     "fileType":"xml",
                     "size":"10"
                  }
               ]
            }
         },
         {
            "id":"2",
            "value":"folder 2",
            "title":"folder 2"
         }
      ]
   }
}
```

most notably is the change from `items` and `children` being just an array of items to being an object that contains additional props for `sortBy` and `filterBy`. now the items will live in an `entries` prop.

the `soryBy` prop represents an array of existing props that will be used as options for the sort dropdown. `sortBy: ['title', 'size'],` in this example `title` and `size` will be used to sort the items. note- there can be multiple attributes to sort by, which is why the value should be an array.

similarly, `filterBy` represents the prop that will be used to filter items. `filterBy: 'fileType',` in this example `fileType` will be used to filter items. as per the spec you can only use a single attribute to filter by, which is why this is not an array.

when there are no values for either `sortBy` or `filterBy` the filter and sort dropdown menus should not be present.

### CODE REVIEW

**Comment by matthewgallo** [comment]
Seems that one of your tests is failing, maybe because of the change in the `items` schema that you mentioned.

**Comment by davidmenendez** [comment]
@matthewgallo you're correct. thank you. fixed.

File: packages/cloud-cognitive/src/components/AddSelect/AddSelect.js [hunk 1]
```diff
@@ -303,4 +323,10 @@ AddSelect.propTypes = {
   title: PropTypes.string,
 };
 
+AddSelect.defaultProps = {
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/AddSelect/AddSelect.js:326]: Should we stick to using a `defaults` object? How come you added back `defaultProps`?

**Review by matthewgallo — APPROVED** [review]
Just had a question about defaultProps vs a defaults object which we've switched most components to use now

File: packages/cloud-cognitive/src/components/AddSelect/AddSelect.js (hunk 1 above)
Comment (davidmenendez) [reply packages/cloud-cognitive/src/components/AddSelect/AddSelect.js:326]: i didn't realize that was the new standard. i can update that going forward

**Pull request merged**